tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/coprocessor/TransactionProcessor.java

```diff
@@ -179,6 +187,13 @@ public void preGetOp(ObserverContext<RegionCoprocessorEnvironment> e, Get get, L
   }
 
   @Override
+  public void prePut(ObserverContext<RegionCoprocessorEnvironment> e, Put put, WALEdit edit, Durability durability)
+    throws IOException {
+    Transaction tx = getFromOperation(put);
+    ensureValidTxLifetime(tx);
+  }
+
+  @Override
```

Comment:
Increment and other operations are non-transactional today. Tephra does not support readless increments.